extensions/amp-consent/0.1/consent-info.js

```diff
@@ -53,31 +64,143 @@ export function getStoredConsentInfo(value) {
+  if (!isEnumValue(CONSENT_ITEM_STATE, newState)) {
+    newState = CONSENT_ITEM_STATE.UNKNOWN;
+  }
+  if (newState == CONSENT_ITEM_STATE.DISMISSED ||
+      newState == CONSENT_ITEM_STATE.UNKNOWN) {
+    return previousState || CONSENT_ITEM_STATE.UNKNOWN;
+  }
+  if (newState == CONSENT_ITEM_STATE.NOT_REQUIRED) {
+    if (previousState && previousState != CONSENT_ITEM_STATE.UNKNOWN) {
+      return previousState;
+    }
+  }
+  return newState;
+}
+
+/**
+ * Compose the value to store to localStorage based on the consentInfo
+ * @param {!ConsentInfoDef} consentInfo
+ * @return {?boolean|Object}
+ */
+export function composeStoreValue(consentInfo) {
+  if (!consentInfo['consentString'] && consentInfo['isDirty'] === undefined) {
+    // TODO: Remove after turn on amp-consent-v2
+    return calculateLegacyStateValue(consentInfo['consentState']);
+  }
+  const obj = Object.create(null);
+  const consentState = consentInfo['consentState'];
+  if (consentState == CONSENT_ITEM_STATE.ACCEPTED) {
+    obj[STORAGE_KEY.STATE] = 1;
+  } else if (consentState == CONSENT_ITEM_STATE.REJECTED) {
+    obj[STORAGE_KEY.STATE] = 0;
+  }
+
+  if (consentInfo['consentString']) {
+    obj[STORAGE_KEY.STRING] = consentInfo['consentString'];
+  }
+
+  if (consentInfo['isDirty'] === true) {
+    obj[STORAGE_KEY.DIRTYBIT] = 1;
+  }
+
+  if (Object.keys(obj) == 0) {
+    return null;
+  }
+
+  return obj;
+}
+
+/**
+ * Convert the consentState to legacy boolean stored value
+ * @param {!CONSENT_ITEM_STATE} consentState
+ * @return {?boolean}
+ */
+export function calculateLegacyStateValue(consentState) {
+  if (consentState == CONSENT_ITEM_STATE.ACCEPTED) {
+    return true;
+  }
+  if (consentState == CONSENT_ITEM_STATE.REJECTED) {
+    return false;
+  }
+  return null;
+}
+
+/**
+ * Deep compare two consentInfo
+ * @param {?ConsentInfoDef} infoA
+ * @param {?ConsentInfoDef} infoB
+ * @return {boolean}
+ */
+export function consentInfoEquals(infoA, infoB) {
+  if (!infoA && !infoB) {
+    return true;
+  }
+  if (infoA && infoB) {
+    const stateEqual = calculateLegacyStateValue(infoA['consentState']) ===
+        calculateLegacyStateValue(infoB['consentState']);
+    const stringEqual =
+        ((infoA['consentString'] || '') === (infoB['consentString'] || ''));
+    const isDirtyEqual = !!infoA['isDirty'] == !!infoB['isDirty'];
+    return stateEqual && stringEqual && isDirtyEqual;
+  }
+  return false;
+}
+
+/**
+ * Helper function to convert stored value to CONSENT_ITEM_STATE value
+ * @param {*} value
+ */
+function convertValueToState(value) {
+  if (value === true || value === 1) {
+    return CONSENT_ITEM_STATE.ACCEPTED;
+  } else if (value === false || value === 0) {
+    return CONSENT_ITEM_STATE.REJECTED;
+  }
+  return CONSENT_ITEM_STATE.UNKNOWN;
+}
+
+/**
+ * Convert the legacy boolean stored value to consentInfo object
+ * @param {boolean} value
+ * @return {!ConsentInfoDef}
+ */
+function getLegacyStoredConsentInfo(value) {
```

Comment:
nit: for readability, please bump this method's position to before the methods it depends on

Reply:
Done. One question, this function also depends on an exported function? Is it preferred to placed exported function above?